ledger/internal/prefetcher/prefetcher.go

```diff
@@ -283,21 +287,60 @@ func (p *accountPrefetcher) prefetch(ctx context.Context) {
 
 	// iterate over the transaction groups and add all their account addresses to the list
 	queue := &tasksQueue
-	for i := range p.groups {
+	for i := range p.txnGroups {
 		task := &groupsReady[i]
-		for j := range p.groups[i] {
-			stxn := &p.groups[i][j]
+		for j := range p.txnGroups[i] {
+			stxn := &p.txnGroups[i][j]
 			switch stxn.Txn.Type {
 			case protocol.PaymentTx:
 				loadAccountsAddAccountTask(&stxn.Txn.Receiver, task, accountTasks, queue)
 				loadAccountsAddAccountTask(&stxn.Txn.CloseRemainderTo, task, accountTasks, queue)
 			case protocol.AssetConfigTx:
+				loadAccountsAddResourceTask(nil, basics.CreatableIndex(stxn.Txn.ConfigAsset), basics.AssetCreatable, task, resourceTasks, queue)
 			case protocol.AssetTransferTx:
+				if !stxn.Txn.AssetSender.IsZero() {
+					loadAccountsAddResourceTask(nil, basics.CreatableIndex(stxn.Txn.XferAsset), basics.AssetCreatable, task, resourceTasks, queue)
+					loadAccountsAddResourceTask(&stxn.Txn.AssetSender, basics.CreatableIndex(stxn.Txn.XferAsset), basics.AssetCreatable, task, resourceTasks, queue)
+				} else {
+					loadAccountsAddResourceTask(&stxn.Txn.Sender, basics.CreatableIndex(stxn.Txn.XferAsset), basics.AssetCreatable, task, resourceTasks, queue)
+					if stxn.Txn.AssetAmount == 0 && (stxn.Txn.AssetReceiver == stxn.Txn.Sender) {
+						// opt in
+						loadAccountsAddResourceTask(nil, basics.CreatableIndex(stxn.Txn.XferAsset), basics.AssetCreatable, task, resourceTasks, queue)
+					}
+				}
+				if !stxn.Txn.AssetReceiver.IsZero() {
+					loadAccountsAddResourceTask(&stxn.Txn.AssetReceiver, basics.CreatableIndex(stxn.Txn.XferAsset), basics.AssetCreatable, task, resourceTasks, queue)
+				}
+				if !stxn.Txn.AssetCloseTo.IsZero() {
+					loadAccountsAddResourceTask(&stxn.Txn.AssetCloseTo, basics.CreatableIndex(stxn.Txn.XferAsset), basics.AssetCreatable, task, resourceTasks, queue)
```

Comment:
If it's a close, we may need to know the creator of the asset (if it's frozen in the sender's account, we check if the closeto is the creator).  Does that mean we have to look up the params?

Reply:
looks like apply logic does not require creator's data

Reply:
we won't need the creator account, but we need the asset params to know who the creator is (I think?)

Reply:
we have a special method for this
```
ok, err := balances.HasAssetParams(source, ct.XferAsset)
```
that checks if the sender is a creator of this asset.

Reply:
I see, you can figure out if an account is the creator by looking at the account local resource record.